lib/Utility/AntiMalware/Providers/ClamAV.php

```diff
@@ -0,0 +1,187 @@
+ * of this program must display Appropriate Legal Notices, as required under
+ * Section 5 of the GNU Affero General Public License version 3.
+ *
+ * In accordance with Section 7(b) of the GNU Affero General Public License version 3,
+ * these Appropriate Legal Notices must retain the display of the "Powered by
+ * SugarCRM" logo and "Supercharged by SuiteCRM" logo. If the display of the logos is not
+ * reasonably feasible for technical reasons, the Appropriate Legal Notices must
+ * display the words "Powered by SugarCRM" and "Supercharged by SuiteCRM".
+ */
+
+namespace SuiteCRM\Utility\AntiMalware\Providers;
+
+
+use Psr\Log\LoggerAwareInterface;
+use Psr\Log\LoggerInterface;
+use SuiteCRM\Exception\Exception;
+use SuiteCRM\Exception\InvalidArgumentException;
+use SuiteCRM\Exception\MalwareFound;
+use SuiteCRM\Interfaces\AntiMalwareFileScanner;
+use SuiteCRM\Utility\Configuration;
+use SuiteCRM\Utility\Paths;
+use SuiteCRM\Utility\SuiteLogger;
+
+class ClamAV implements AntiMalwareFileScanner, LoggerAwareInterface
+{
+    /**
+     * @var bool $enabled
+     */
+    private $enabled = false;
+
+    /**
+     * @var string $path
+     */
+    private $path = '/usr/bin/clamscan';
+
+    /**
+     * @var string $options
+     */
+    private $options = '--stdout --infected';
+
+    /**
+     * @var  LoggerInterface
+     */
+    private $logger;
+
+    public function __construct()
+    {
+        $globalConfig = new Configuration();
+        $configuration =
+            isset($globalConfig['anti_malware_scanners'][self::class])
+                ? $globalConfig['anti_malware_scanners'][self::class]
+                : null;
+
+        if(!empty($configuration)) {
+           $this->enabled = isset($configuration['enabled']) ?  $configuration['enabled'] : $this->enabled;
+           $this->path = isset($configuration['path']) ?  $configuration['path'] : $this->path;
+           $this->options = isset($configuration['options']) ?  $configuration['options'] : $this->options;
+        }
+
+        $this->logger = new SuiteLogger();
+
+    }
+
+    /**
+     * Is the Anti Malware Scanner installed on the system
+     * @return bool true === yes, false === no
+     */
+    public function isAntiMalwareScannerAvailable()
+    {
+        if ($this->isEnabled()) {
+            return file_exists($this->getExecutablePath());
+        }
+
+        return false;
+    }
+
+    /**
+     * @param string $path - location in the file system
+     * @throws InvalidArgumentException
+     * @throws MalwareFound
+     */
+    public function scanPathForMalware($path)
+    {
+        if (!is_string($path)) {
+            throw new InvalidArgumentException(
+                '[ClamAV][scanPathForMalware] $path should be a string'
+            );
+
+        }
+
+        if (!file_exists($this->getExecutablePath())) {
+            throw new InvalidArgumentException(
+                '[ClamAV][scanPathForMalware] $path not found'
```

Comment:
$path will not be replaced here, it might be more useful for debugging to have the value of this variable?

Same goes for the similar exception in SophosAntiVirus.php

Reply:
I actually wasn't intending this to be replaced by the value. I guess we could though :+1:

Reply:
OK so it looks like I have overlooked something here. So have added an extra check here. Thank you for raising this. :D